accepted/0000-yarn-tasks.md

```diff
@@ -0,0 +1,103 @@
+- Start Date: 2018-11-21
+- RFC PR:
+- Yarn Issue: https://github.com/yarnpkg/yarn/issues/5773
+
+# Summary
+
+Yarn CLI command `yarn run` will be able to orchestrate various npm scripts, allowing them to run in parallel by default and matching any glob patterns.
+
+# Motivation
+
+Many build processes involve Grunt, Gulp, or other task runners that have great use cases but also require separate plugins, configuration files, and understanding of the API. `yarn run` would remove the need for installing these often global dpendencies and lower maintenance costs by providing a simpler interface defined in package.json. This would also cut down on the script run times and would be useful with workspaces as mentioned in https://github.com/yarnpkg/yarn/issues/6689.
+
+Feature Requirements:
+
+- Be able to run multiple commands in parallel across platform
+- Show and differentiate console output for each command
+- Compatible with existing behavior of `npm run` command
+
+Prior Art:
+
+- [npm-run-all](https://www.npmjs.com/package/npm-run-all)
+- [gulp](https://gulpjs.com/)
+
+# Detailed design
+
+## Simple use case
+
+// In `package.json`
+
+```js
+{
+    "scripts": {
+        "lint": "eslint .",
+        "test": "jest"
+    }
+    "tasks": {
+        "verify": ["lint", "test"]
+    }
+}
+```
+
+`yarn run verify` would run lint and test commands in parallel
+
+## Advanced use case - pattern matching
+
+```js
+{
+    "scripts": {
+        "lint": "eslint .",
+        "test:unit": "jest __tests__",
+        "test:integration": "jest __integration__"
+    }
+    "tasks": {
+        "verify": ["lint", "test**"]
+    }
+}
+```
+
+`yarn run verify` would run lint, test:unit, and test:integration commands in parallel
+
+## Advanced use case - nested tasks
+
+```js
+{
+    "scripts": {
+        "build": "webpack",
+        "lint": "eslint .",
+        "test": "jest",
+    }
+    "tasks": {
+        "verify": ["lint", "test"],

```

Comment:
One slight worry is that `lint-staged` also uses an array to specify git pre-commit tasks but they use it to mean serial semantics.

Reply:
You _could_ make the tasks an object instead of an array to signify that there's no defined order, but that would be a pain to write.

How big of a deal do you think the issue with `lint-staged` is?

Reply:
I agree that an object would be a pain. The `lint-staged` issue is something that can be handled by documentation so I don't see it as a blocking issue. Of course, there's nothing to say that `lint-staged` couldn't/shouldn't support parallel tasks and they may adopt whatever syntax `yarn` chooses.